activation/activation.go

```diff
@@ -483,58 +520,64 @@ func (b *Builder) Coinbase() types.Address {
 	return b.coinbaseAccount
 }
 
-// MinGas [...].
-func (b *Builder) MinGas() uint64 {
-	panic("not implemented")
-}
-
-// SetMinGas [...].
-func (b *Builder) SetMinGas(value uint64) {
-	panic("not implemented")
-}
-
-func getNIPostKey() []byte {
-	return []byte("NIPost")
-}
-
-func (b *Builder) storeChallenge(ch *types.NIPostChallenge) error {
-	bts, err := codec.Encode(ch)
+func (b *Builder) loadChallenge() error {
+	nipost, err := store.GetNIPostChallenge(b.cdb)
 	if err != nil {
-		return fmt.Errorf("serialize NIPost challenge: %w", err)
+		return err
 	}
+	b.challenge = nipost
+	return nil
+}
 
-	if err := niposts.Add(b.cdb, getNIPostKey(), bts); err != nil {
-		return fmt.Errorf("put NIPost challenge to database: %w", err)
+func (b *Builder) discardChallenge() {
+	b.challenge = nil
+	b.pendingATX = nil
+	if err := store.ClearNIPostChallenge(b.cdb); err != nil {
+		b.log.Error("failed to discard NIPost challenge: %w", err)
 	}
-
-	return nil
 }
 
-func (b *Builder) loadChallenge() error {
-	bts, err := niposts.Get(b.cdb, getNIPostKey())
-	if err != nil {
-		return fmt.Errorf("get NIPost challenge from store: %w", err)
+func (b *Builder) getCommitmentAtx(ctx context.Context) (*types.ATXID, error) {
+	if b.commitmentAtx != nil {
+		return b.commitmentAtx, nil
 	}
 
-	if len(bts) > 0 {
-		var tp types.NIPostChallenge
-		if err = codec.Decode(bts, &tp); err != nil {
-			return fmt.Errorf("parse NIPost challenge: %w", err)
-		}
+	syncedCh := b.syncer.RegisterChForSynced(ctx)
```

Comment:
why do we need the node to be synced to get the commitment ATX?

Reply:
Because the commitment ATX selected by a node should be the one with the highest height it has seen before starting to initialise PoST.

If we don't sync here, the selected commitment ATX will always be the golden ATX.

Reply:
to clarify. 
- for genesis miners (miners that start smeshing from epoch 0), they will have have golden ATX for commitment ATX. 
- for semshers that join at epoch 1, it's possible that some of them have golden ATX for commitment ATX, and some of them have valid ATX for commitment ATX.
- for smeshers that join at/after epoch 2, they should all have valid ATX for commitment ATX

do i understand correctly?

Reply:
After the discussion with Tal today: 

- The commitment ATX should be the ATX with the maximum hight the node as seen when it starts smeshing. This is not validated / enforced by other nodes.
- A node can choose any ATX for its commitment as it wants, but choosing the ATX with the highest available hight gives the node the highest identity tick height.
- This identity tick height is used in case of an attack against the tortoise beacon where honest parties prioritize IDs with the highest tick numbers.

Reply:
right. with Tal's info, can you confirm the effective behavior i listed if a node were to follow the protocol?

if what i stated is right. then all genesis miners will have empty commitment ATX and will be de-prioritized in beacon protocol.